live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/storage/StorageClientIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.storage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageClient;
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageConfig;
+import org.alfresco.hxi_connector.live_ingester.utils.DockerTags;
+
+@SpringBootTest(classes = {LocalStorageConfig.class, SignedStorageClient.class})
+@ActiveProfiles({"test"})
+@Testcontainers
+class StorageClientIntegrationTest
+{
+    private static final String LOCALSTACK_IMAGE = "localstack/localstack";
+    private static final String LOCALSTACK_TAG = DockerTags.getOrDefault("localstack.tag", "3.0.2");
+    private static final String BUCKET_NAME = "test-hxinsight-bucket";
+    private static final String OBJECT_KEY = "dummy.txt";
+    private static final String OBJECT_CONTENT = "dummy's file dummy content";
+    private static final String OBJECT_CONTENT_TYPE = "plain/text";
+
+    @Container
+    @SuppressWarnings("PMD.FieldNamingConventions")
+    private static final LocalStackContainer localStack = new LocalStackContainer(DockerImageName.parse(LOCALSTACK_IMAGE).withTag(LOCALSTACK_TAG));
+
+    @Autowired
+    private LocalStorageClient testS3Storage;
+    @Autowired
+    private StorageClient storageClient;
+
+    @BeforeAll
+    static void beforeAll() throws IOException, InterruptedException
+    {
+        localStack.execInContainer("awslocal", "s3api", "create-bucket", "--bucket", BUCKET_NAME);
+    }
+
+    @Test
+    void testUpload() throws IOException
+    {
+        List<String> freshBucketContent = testS3Storage.listBucketContent(BUCKET_NAME);
+        assertThat(freshBucketContent)
+                .isNotNull()
+                .isEmpty();
+
+        // given
+        try (InputStream fileInputStream = new ByteArrayInputStream(OBJECT_CONTENT.getBytes()))
```

Comment:
I am not sure as I never used it but when I was checking lombok features I found this annotation https://projectlombok.org/features/Cleanup, dyt we can use it here?